Scripts/Process/Component/GameObjectSourceTargetProcessor.cs

```diff
@@ -0,0 +1,187 @@
+        /// </summary>
+        [Header("Entity Settings"), Tooltip("A GameObject collection of sources to apply data from.")]
+        public List<GameObject> sources = new List<GameObject>();
+        /// <summary>
+        /// Allows to optionally determine which sources should be processed based on the set rules.
+        /// </summary>
+        [Tooltip("Allows to optionally determine which sources should be processed based on the set rules.")]
+        public RuleContainer sourceValidity;
+        /// <summary>
+        /// A <see cref="GameObject"/> collection of targets to apply data to.
+        /// </summary>
+        [Tooltip("A GameObject collection of targets to apply data to.")]
+        public List<GameObject> targets = new List<GameObject>();
+        /// <summary>
+        /// Allows to optionally determine which targets should be processed based on the set rules.
+        /// </summary>
+        [Tooltip("Allows to optionally determine which targets should be processed based on the set rules.")]
+        public RuleContainer targetValidity;
+        #endregion
+
+        /// <summary>
+        /// The current sources collection index.
+        /// </summary>
+        public int CurrentSourcesIndex
+        {
+            get { return _currentSourcesIndex; }
+            set { _currentSourcesIndex = sources.GetWrappedAndClampedIndex(value); }
+        }
+        private int _currentSourcesIndex;
+
+        /// <summary>
+        /// The current targets collection index.
+        /// </summary>
+        public int CurrentTargetsIndex
+        {
+            get { return _currentTargetsIndex; }
+            set { _currentTargetsIndex = targets.GetWrappedAndClampedIndex(value); }
+        }
+        private int _currentTargetsIndex;
+
+        /// <inheritdoc />
+        public override void Process()
+        {
+            if (!isActiveAndEnabled)
+            {
+                return;
+            }
+
+            ApplySourcesToTargets(sources, targets);
+        }
+
+        /// <summary>
+        /// Adds the given source to the sources collection.
+        /// </summary>
+        /// <param name="source">The source to add.</param>
+        public virtual void AddSource(GameObject source)
+        {
+            sources.Add(source);
+        }
+
+        /// <summary>
+        /// Removes the given source from the sources collection.
+        /// </summary>
+        /// <param name="source">The source to remove.</param>
+        public virtual void RemoveSource(GameObject source)
+        {
+            sources.Remove(source);
+        }
+
+        /// <summary>
+        /// Sets the given source at the given index of the sources collection.
+        /// </summary>
+        /// <param name="index">The index to set the source at.</param>
+        /// <param name="source">The source to set.</param>
+        public virtual void SetSource(int index, GameObject source)
```

Comment:
`sources` is `public` and a mutable `List`, so is this needed?

Users can get access to the list themselves and mutate it that way and UnityEvents don't allow being wired up to a method taking more than one argument.

Reply:
This just tidies up the `SetSourceAtCurrentIndex` method to simply call `SetSource`

But we could ditch this method and move the internals to `SetSourceAtCurrentIndex`